socorro/processor/breakpad_transform_rules.py

```diff
@@ -459,3 +490,153 @@ def _predicate(
         return 'create_dump_lookup' in raw_crash
 
 
+#==============================================================================
+class BreakpadStackwalkerRule2015(ExternalProcessRule):
+
+    required_config = Namespace()
+    required_config.add_option(
+        name='symbols_url',
+        doc='url of the symbol server',
+        default="https://localhost"
+    )
+    required_config.command_line = change_default(
+        ExternalProcessRule,
+        'command_line',
+        'timeout -s KILL 30 {command_pathname} '
+        '--raw-json {raw_crash_pathname} '
+        '--symbols-url {symbols_url} '
+        '--symbols-cache {symbol_cache_path} '
+        '{dump_file_pathname} '
+        '2>/dev/null'
+    )
+    required_config.command_pathname = change_default(
+        ExternalProcessRule,
+        'command_pathname',
+        '/data/socorro/stackwalk/bin/stackwalker',
+    )
+    required_config.add_option(
+        'symbol_cache_path',
+        doc='the path where the symbol cache is found (quote path with '
+        'embedded spaces)',
+        default='/mnt/socorro/symbols',
+    )
+    required_config.add_option(
+        'temporary_file_system_storage_path',
+        doc='a path where temporary files may be written',
+        default='/tmp',
```

Comment:
@rhelmer: running tempfile.gettempdir() will just set the temporary
directory location to a constant at import time.  That constant can be
overridden in the same manner as any other constant that might be put
there.

```gettempdir``` doesn't create a new temporary directory, it just search a
set of known typical temp directory locations.  It'll return the first one
that it finds for which the current user has write permission.


On Tue, Feb 3, 2015 at 10:45 AM, Robert Helmer <notifications@github.com>
wrote:

> In socorro/processor/breakpad_transform_rules.py
> <https://github.com/mozilla/socorro/pull/2614#discussion_r24027616>:
>
> > +    )
> > +    required_config.command_pathname = change_default(
> > +        ExternalProcessRule,
> > +        'command_pathname',
> > +        '/data/socorro/stackwalk/bin/stackwalker',
> > +    )
> > +    required_config.add_option(
> > +        'symbol_cache_path',
> > +        doc='the path where the symbol cache is found (quote path with '
> > +        'embedded spaces)',
> > +        default='/mnt/socorro/symbols',
> > +    )
> > +    required_config.add_option(
> > +        'temporary_file_system_storage_path',
> > +        doc='a path where temporary files may be written',
> > +        default='/tmp',
>
> @peterbe <https://github.com/peterbe> can I use tempfile.gettempdir()
> here? How do I override it at runtime with configman?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/socorro/pull/2614/files#r24027616>.
>